docs/manual/howto/git.rst

```diff
@@ -0,0 +1,104 @@
+   clear, consult other sources until you are confident you know what you are
+   doing.
+
+I want to try out a feature somebody is working on
+==================================================
+If you see a pull request on `GitHub <https://www.github.com/qtile/qtile/pulls>`_
+that you want to try out, have a look at the line where it says::
+
+  user wants to merge n commits into qtile:master from user:branch
+
+Right now you probably have one *remote* from which you can fetch changes, the
+``origin``. If you cloned ``qtile/qtile``, ``git remote show origin`` will spit
+out the *upstream* url. If you cloned your fork, ``origin`` points to it and you
+probably want to ``git remote add upstream https://www.github.com/qtile/qtile``.
+To try out somebody's work, you can add his fork as a new remote::
+
+  git remote add user https://www.github.com/user/qtile
+
+where you fill in the username from the line we asked you to search for before.
+Then you can load data from that remote with ``git fetch`` and then ultimately
+check out the branch with ``git checkout user/branch``.
+
+.. note:: Having the feature branch checked out doesn't mean that it is
+   installed and will be loaded when you restart qtile. You might still need to
+   install it with ``pip``.
+
+I committed changes and the tests failed
+========================================
+
+You can easily change your last commit: After you have done your work,
+``git add`` everything you need and use ``git commit --amend`` to change your
+last commit. This causes the git history of your local clone to be diverged from
+your fork on GitHub, so you need to force-push your changes with::
+
+  git push -f your_username
+
+I was told to rebase my work
+============================
+
+If *upstream/master* is changed and you happened to change the same files as the
+commits that were added upstream, you should rebase your work onto the most
+recent *upstream/master*. Checkout your master, pull from *upstream*, checkout
+your branch again and then rebase it::
+
+  git checkout master
+  git pull upstream/master
+  git checkout feature-branch
+  git rebase upstream/master
+
+You will be asked to solve conflicts where your diff cannot be applied with
+confidence to the work that was pushed upstream. If that is the case, open the
+files in your text editor and resolve the conflicts manually. You possibly need
+to ``git rebase --continue`` after you have resolved conflicts for one commit if
+you are rebasing multiple commits.
+
+Note that the above doesn't work if you didn't create a branch. In that case you
+will find guides elsewhere to fix this problem, ideally by creating a branch and
+resetting your master branch to where it should be.
+
+I was told to squash some commits
+=================================
+
+If you introduce changes in one commit and replace them in another, you are told
+to squash these changes into one single commit without the intermediate step::
+
+  git rebase -i
```

Comment:
Running this command actually gives me an error. I think you usually have to say what you're rebasing against.

Typically, when squashing commits, I'd run something like:
```
git rebase -i <branch-name>~4
```
to show me the last 4 commits on my branch.

Reply:
It probably depends on what you have checked out. Wildly guessing I'd say `git rebase -i` is a shortcut for `git rebase -i master` assuming you're on a feature branch. Not completely certain about that though...